crates/app/src/handle.rs

```diff
@@ -55,7 +55,7 @@ pub async fn handle(mut req: Request<Body>) -> impl IntoResponse {
     })?;
     assert_eq!(extensions.insert(repo), None, "duplicate repository name");
 
-    let mut tail = tail.split_terminator('/');
+    let mut tail = tail.trim_end_matches('/').splitn(4, '/');
```

Comment:
```suggestion
    let mut tail = tail.splitn(4, '/');
```

If we get `"_tag/0.1.0/tree/a//////` we want to handle that case correctly (most probably fail) and not assume `a`
`TreePath::FromStr` implementation correctly handles `/` anyway

Reply:
Can you clarify what you want to see? Do you just want the previous behavior, where it effectively stripped a single trailing `/` rather than all the trailing `/`s? What behavior do we actually want here, and why?

Reply:
I think simply removing the slash trimming as suggested in the diff above should work just fine, otherwise we may consider trimming the very last slash.

Reply:
Sure thing, I'll do that. Although, regarding point 2, I don't see why it would be incompatible with the splitn approach, we would just need to have a preprocessing step that normalizes the string to coalesce adjacent slashes.

Reply:
> Sure thing, I'll do that. Although, regarding point 2, I don't see why it would be incompatible with the splitn approach, we would just need to have a preprocessing step that normalizes the string to coalesce adjacent slashes.

it would not be possible to "simply" skip the empty elements in the iterator, which would be a trivial change with `split_terminator`